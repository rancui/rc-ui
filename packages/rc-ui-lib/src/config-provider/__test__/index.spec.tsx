import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Icon from '../../icon';
import ConfigProvider from '..';

describe('ConfigProvider', () => {
  afterEach(() => {
    cleanup();
  });

  it('when using themeVars prop', () => {
    const themeVars = {
      color: 'red',
    };
    const { container } = render(
      <ConfigProvider themeVars={themeVars}>
        <Icon name="success" />
      </ConfigProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('when using iconPrefix prop', () => {
    const { container } = render(
      <ConfigProvider iconPrefix="my-icon">
        <Icon name="success" />
      </ConfigProvider>,
    );
    expect(container.querySelector('i.my-icon-success')).toBeTruthy();
  });
});
