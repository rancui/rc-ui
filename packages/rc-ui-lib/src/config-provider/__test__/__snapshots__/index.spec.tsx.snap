// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigProvider when using themeVars prop 1`] = `
<div>
  <div
    style="--rc-color: red;"
  >
    <i
      class="rc-badge__wrapper van-icon van-icon-success"
    />
  </div>
</div>
`;
